base/test/sound_record_tests.cpp

```diff
@@ -0,0 +1,94 @@
+#include "stdafx.h"
+#include <boost/test/unit_test.hpp>
+#include "FrameMetadata.h"
+#include "FrameMetadataFactory.h"
+#include "Frame.h"
+#include "Logger.h"
+#include "test_utils.h"
+#include "PipeLine.h"
+#include "FileWriterModule.h"
+#include "SoundRecord.h"
+#include "ExternalSinkModule.h"
+#include "Module.h"
+
+BOOST_AUTO_TEST_SUITE(sound_record_tests)
+
+BOOST_AUTO_TEST_CASE(recordMono, *boost::unit_test::disabled())
+{
+    // Manual test, listen to the file on audacity to for sanity check
+    Logger::setLogLevel(boost::log::trivial::severity_level::info);
+
+    auto time_to_run = Test_Utils::getArgValue("s", "10");
+    auto n_seconds = atoi(time_to_run.c_str());
+
+    SoundRecordProps sourceProps(48000,1,0,200);
+    auto source = boost::shared_ptr<Module>(new SoundRecord(sourceProps));
+
+    auto outputFile = boost::shared_ptr<Module>(new FileWriterModule(FileWriterModuleProps("./data/AudiotestMono.wav", true)));
+    source->setNext(outputFile);
+
+    PipeLine p("test");
+    p.appendModule(source);
+    p.init();
+    p.run_all_threaded();
+    boost::this_thread::sleep_for(boost::chrono::seconds(n_seconds));
+    p.stop();
+    p.term();
+    p.wait_for_all();
```

Comment:
Since we are using a wav file and we know the chanels and samplingrate, we can estimate the expected byteSize for the output file and assert that it is in the right ball park let's say +/- 2% variability allowed. it will be one step closer to automated test. so if the file came to be a grossly wrong size, no need to play it back.

Reply:
OK one comment which we can do it as an improvement after this merge, **the tests should always create any litter generated...** e.g.  Audiotest*.wav files generated here will be left in the developers workspace and sometimes the test will be fooled because of the litter. e.g. woru writing fails but a past copy of file exists in the same workspace. 

The solution is to add a destructor to the test  ~sound_record_tests() which deletes the files in a tolerant fashion so that if the file does not exit it does not crash and move to the next file.